### PR TITLE
[HttpFoundation] remove deprecated features

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
looks like we need #22873 first

**Comment by nicolas-grekas** [comment]
so, InlineFragmentRenderer 4 still conditionally calls getTrustedHeaderName for BC for 3.4?

File: src/Symfony/Component/HttpFoundation/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,23 @@
 CHANGELOG
 =========
 
+4.0.0
+-----
+
+ * the `Request::setTrustedHeaderName()` and `Request::getTrustedHeaderName()`
+   methods have been removed
+ * the `Request::HEADER_CLIENT_IP` constant has been removed, use
+   `Request::HEADER_X_FORWARDED_FOR` instead
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/CHANGELOG.md]: you should configure your line breaks at 120, this is cutting too early to me

Comment (xabbuh) [inline src/Symfony/Component/HttpFoundation/CHANGELOG.md]: IIRC @fabpot once talked about a limit of 80 characters in changelog files

**Comment by xabbuh** [comment]
> so, InlineFragmentRenderer 4 still conditionally calls getTrustedHeaderName for BC for 3.4?

Ah no, that was not intended. Should be fixed.

**Comment by xabbuh** [comment]
Status: Needs Review

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Comment by c960657** [comment]
The documentation for getClientIp(), getPort(), isSecure() and getHost() includes the following paragraph:

> If your reverse proxy uses a different header name than "X-Forwarded-Xxx", configure it via via the $trustedHeaderSet argument of the Request::setTrustedProxies() method instead.

This feature is no longer supported, so this must be removed from the documentation.

**Pull request merged**